common/src/main/java/com/wynntils/screens/maps/WaypointCreationScreen.java

```diff
@@ -0,0 +1,1303 @@
+            labelColorInput.setTextBoxInput(oldAttributes.labelColor().toHexString());
+        } else if (labelColorInput.getTextBoxInput().isEmpty()) {
+            labelColorInput.setTextBoxInput("#FFFFFF");
+        }
+        // endregion
+
+        // region Icon
+        availableIcons.clear();
+        availableIcons.addAll(Services.MapData.getIcons().toList());
+
+        if (firstSetup && oldAttributes != null) {
+            String oldIconId = oldAttributes.iconId();
+
+            if (oldIconId.equals(MapIcon.NO_ICON_ID)) {
+                useIcon = false;
+            } else {
+                Optional<MapIcon> oldMapIcon = availableIcons.stream()
+                        .filter(mapIcon -> mapIcon.getIconId().equals(oldIconId))
+                        .findFirst();
+
+                oldMapIcon.ifPresent(oldIcon -> selectedIcon = oldIcon);
+
+                iconScrollOffset = availableIcons.indexOf(selectedIcon);
+
+                iconId = oldIconId;
+            }
+        }
+
+        iconCheckbox = new WynntilsCheckbox(
+                (int) dividedWidth,
+                (int) (dividedHeight * 25),
+                20,
+                20,
+                Component.translatable("screens.wynntils.poiCreation.icon"),
+                useIcon,
+                (int) (dividedWidth * 6),
+                (b -> toggleIcon()));
+        this.addRenderableWidget(iconCheckbox);
+
+        previousIconButton = new Button.Builder(Component.literal("<"), (button) -> {
+                    if (iconScrollOffset - 1 < 0) {
+                        iconScrollOffset = availableIcons.size() - 1;
+                    } else {
+                        iconScrollOffset--;
+                    }
+
+                    populateIcons();
+                })
+                .pos((int) (dividedWidth * 8), (int) (dividedHeight * 25))
+                .size(20, 20)
+                .build();
+        this.addRenderableWidget(previousIconButton);
+
+        nextIconButton = new Button.Builder(Component.literal(">"), (button) -> {
+                    if (iconScrollOffset + 1 >= availableIcons.size()) {
+                        iconScrollOffset = 0;
+                    } else {
+                        iconScrollOffset++;
+                    }
+
+                    populateIcons();
+                })
+                .pos((int) (dividedWidth * 20), (int) (dividedHeight * 25))
+                .size(20, 20)
+                .build();
+        this.addRenderableWidget(nextIconButton);
+
+        addCustomIconButton = new Button.Builder(
+                        Component.translatable("screens.wynntils.poiCreation.addCustomIcon"),
+                        (button) -> McUtils.mc().setScreen(CustomWaypointIconScreen.create(this)))
+                .pos((int) (dividedWidth * 8) + 20, (int) (dividedHeight * 25) + 20)
+                .size((int) (dividedWidth * 10), 20)
+                .build();
+        this.addRenderableWidget(addCustomIconButton);
```

Comment:
This button could be the same width as the displayed icons itself, or at least, center aligned with those buttons.